docs/src/main/asciidoc/config-mappings.adoc

```diff
@@ -0,0 +1,458 @@
+the default behaviours of ConfigxMapping interfaces.
+
+=== Examples
+
+==== Simple Config Mapping
+[source,java]
+----
+interface Server {
+    String host();
+
+    int port();
+}
+----
+
+==== Nested Config Mapping
+
+[source,java]
+----
+interface ServerSub {
+    @WithParentName
+    ServerSubHostAndPort subHostAndPort();
+
+    @WithParentName
+    ServerSubName subName();
+}
+
+interface ServerSubHostAndPort {
+    String host();
+
+    int port();
+}
+
+interface ServerSubName {
+    String name();
+}
+----
+
+==== Types Config Mapping
+
+[source,java]
+----
+public interface SomeTypes {
+    @WithName("int")
+    int intPrimitive();
+
+    @WithName("int")
+    Integer intWrapper();
+
+    @WithName("long")
+    long longPrimitive();
+
+    @WithName("long")
+    Long longWrapper();
+
+    @WithName("float")
+    float floatPrimitive();
+
+    @WithName("float")
+    Float floatWrapper();
+
+    @WithName("double")
+    double doublePrimitive();
+
+    @WithName("double")
+    Double doubleWrapper();
+
+    @WithName("char")
+    char charPrimitive();
+
+    @WithName("char")
+    Character charWrapper();
+
+    @WithName("boolean")
+    boolean booleanPrimitive();
+
+    @WithName("boolean")
+    Boolean booleanWrapper();
+}
+----
+
+==== Optional Config Mapping
+
+[source,java]
+----
+public interface Optionals {
+    Optional<Server> server();
+
+    Optional<String> optional();
+
+    @WithName("optional.int")
+    OptionalInt optionalInt();
+}
+----
+
+==== Collections Config Mapping
+
+[source,java]
+----
+public interface CollectionTypes {
+    @WithName("strings")
+    List<String> listStrings();
+
+    @WithName("ints")
+    List<Integer> listInts();
+}
+
+public interface Maps {
+    @WithParentName
+    Map<String, Server> server();
+
+    Map<String, String> anotherServer();
+}
+
+public interface ServerCollections {
+    List<Environment> environments();
+
+    interface Environment {
+        String name();
+
+        List<App> apps();
+
+        interface App {
+            String name();
+
+            List<String> services();
+
+            Optional<List<String>> databases();
+        }
+    }
+}
+----
+
+==== Defaults Config Mapping
+
+[source,java]
+----
+public interface Defaults {
+    @WithDefault("foo")
+    String foo();
+
+    @WithDefault("bar")
+    String bar();
+}
+----
+
+==== Converter Config Mapping
+
+[source,java]
+----
+public interface Converters {
+    @WithConverter(FooBarConverter.class)
+    String foo();
+}
+
+public static class FooBarConverter implements Converter<String> {
+    @Override
+    public String convert(final String value) {
+        return "bar";
+    }
+}
+----
+
+[[config-properties]]
+== [.line-through]#@ConfigProperties# (Deprecated)
+
+IMPORTANT: This feature will be removed soon, please update your code base and use @ConfigMapping instead.
+
+As an alternative to injecting multiple related configuration values is to use the
```

Comment:
`As an alternative to... is` sounds weird, I would have either expected `An alternative` or a second part to this sentence.